Source/Core/Common/EnumMap.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <type_traits>
+
+#include "Common/TypeUtils.h"
+
+template <std::size_t position, std::size_t bits, typename T, typename StorageType>
+struct BitField;
+
+// A type that allows lookup of values associated with an enum as the key.
+// Designed for enums whose numeric values start at 0 and increment continuously with few gaps.
+template <typename V, auto last_member>
```

Comment:
This should be placed in the Common namespace.

Maybe also a unit test or two as well in the tests.

Reply:
Moved to `Common`.  I don't think there are any tests I can reasonably write; the `EnumFormatter` test covers it pretty well and most of the other logic is something that would produce a compile-time error when violated (which I don't think can be handled in a test).